tests/benchmarks/peripheral_stress_test/src/i2s_thread.c

```diff
@@ -0,0 +1,201 @@
+{
+	int ret;
+	struct i2s_config i2s_cfg;
+
+	i2s_cfg.word_size = 16U;
+	i2s_cfg.channels = 2U;
+	i2s_cfg.format = I2S_FMT_DATA_FORMAT_I2S;
+	i2s_cfg.frame_clk_freq = FRAME_CLK_FREQ;
+	i2s_cfg.block_size = BLOCK_SIZE;
+	i2s_cfg.timeout = TIMEOUT;
+
+	if (dir == I2S_DIR_TX) {
+		/* Configure the Transmit port as Master */
+		i2s_cfg.options = I2S_OPT_FRAME_CLK_MASTER
+				| I2S_OPT_BIT_CLK_MASTER;
+	} else if (dir == I2S_DIR_RX) {
+		/* Configure the Receive port as Slave */
+		i2s_cfg.options = I2S_OPT_FRAME_CLK_SLAVE
+				| I2S_OPT_BIT_CLK_SLAVE;
+	} else { /* dir == I2S_DIR_BOTH */
+		i2s_cfg.options = I2S_OPT_FRAME_CLK_MASTER
+				| I2S_OPT_BIT_CLK_MASTER;
+	}
+
+	if (dir == I2S_DIR_TX || dir == I2S_DIR_BOTH) {
+		i2s_cfg.mem_slab = &tx_mem_slab;
+		ret = i2s_configure(dev_i2s, I2S_DIR_TX, &i2s_cfg);
+		if (ret < 0) {
+			LOG_ERR("Failed to configure I2S TX stream (%d)", ret);
+			return -1;
+		}
+	}
+
+	if (dir == I2S_DIR_RX || dir == I2S_DIR_BOTH) {
+		i2s_cfg.mem_slab = &rx_mem_slab;
+		ret = i2s_configure(dev_i2s, I2S_DIR_RX, &i2s_cfg);
+		if (ret < 0) {
+			LOG_ERR("Failed to configure I2S RX stream (%d)", ret);
+			return -1;
+		}
+	}
+
+	return 0;
+}
+
+
+static void tx_block_write(const struct device *dev_i2s, int att, int err)
+{
+	char tx_block[BLOCK_SIZE];
+	int ret;
+
+	fill_buf((uint16_t *)tx_block, att);
+	ret = i2s_buf_write(dev_i2s, tx_block, BLOCK_SIZE);
+	if (ret != err) {
+		LOG_ERR("i2s_buf_write failed (%d)", ret);
+	}
+}
+
+static void rx_block_read(const struct device *dev_i2s, int att)
+{
+	char rx_block[BLOCK_SIZE];	/* Data is received but left unalyzed */
+	size_t rx_size;
+	int ret;
+
+	ret = i2s_buf_read(dev_i2s, rx_block, &rx_size);
+	if (ret < 0 || rx_size != BLOCK_SIZE) {
+		LOG_ERR("i2s_buf_read() failed (%d)", ret);
+	} else {
+		LOG_DBG("i2s_buf_read() received %d", rx_size);
+	}
+}
+
+/* I2S thread */
+static void i2s_thread(void *arg1, void *arg2, void *arg3)
+{
+	ARG_UNUSED(arg1);
+	ARG_UNUSED(arg2);
+	ARG_UNUSED(arg3);
+
+	int ret;
+
+	atomic_inc(&started_threads);
+
+	if (!device_is_ready(dev_i2s)) {
+		LOG_ERR("Device %s is not ready.", dev_i2s->name);
+		atomic_inc(&completed_threads);
+		return;
+	}
+
+	/* Configure I2S */
+	ret = configure_stream(dev_i2s, I2S_DIR_BOTH);
+	if (ret < 0) {
+		atomic_inc(&completed_threads);
+		return;
+	}
+
+	for (int i = 0; i < I2S_THREAD_COUNT_MAX; i++) {
+		/* Drop??? previous communication */
+		ret = i2s_trigger(dev_i2s, I2S_DIR_BOTH, I2S_TRIGGER_DROP);
+		if (ret < 0) {
+			LOG_ERR("i2s_trigger(I2S_TRIGGER_DROP) TX/RX failed (%d)", ret);
+		}
+
+		/* Prefill TX queue */
+		tx_block_write(dev_i2s, 0, 0);
+		LOG_DBG("TX %d->OK", 1);
+
+		tx_block_write(dev_i2s, 1, 0);
+		LOG_DBG("TX %d->OK", 2);
+
+		ret = i2s_trigger(dev_i2s, I2S_DIR_BOTH, I2S_TRIGGER_START);
+		if (ret < 0) {
+			LOG_ERR("i2s_trigger(I2S_TRIGGER_START) TX/RX failed (%d)", ret);
+		}
+
+		rx_block_read(dev_i2s, 0);
+		LOG_DBG("RX %d<-OK", 1);
+
+		tx_block_write(dev_i2s, 2, 0);
```

Comment:
handle error from this call and exit if such occurs

Reply:
fixed